integration-test/src/main/java/com/sequenceiq/it/cloudbreak/cloud/v4/azure/AzureCloudProvider.java

```diff
@@ -252,15 +241,26 @@ public DistroXNetworkTestDto network(DistroXNetworkTestDto network) {
 
     @Override
     public EnvironmentNetworkTestDto network(EnvironmentNetworkTestDto network) {
-        return network.withSubnetIDs(getSubnetIds())
-                .withAzure(environmentNetworkParameters());
+        boolean privateEndpointEnabled = isPrivateEndpointEnabled(network);
+        return network.withSubnetIDs(getSubnetIds(privateEndpointEnabled))
+                .withAzure(environmentNetworkParameters(privateEndpointEnabled));
     }
 
-    private EnvironmentNetworkAzureParams environmentNetworkParameters() {
+    private boolean isPrivateEndpointEnabled(EnvironmentNetworkTestDto network) {
```

Comment:
Based on our discussion, I think this whole `isPrivateEndpointEnabled` method is not needed anymore, because of it seems duplicates the `serviceEndpoint` method here. Furthermore we are going to introduce this parameter directly in the `withServiceEndpoints` at test level via `EnvironmentNetworkTestDto` (you can see below). So here we can use jsut the `environmentNetworkAzureParams.setDatabasePrivateDnsZoneId(network.getDatabasePrivateDnsZoneId());` with any condition.